packages/opentelemetry-tracing/src/BasicTracerProvider.ts

```diff
@@ -112,6 +112,22 @@ export class BasicTracerProvider implements TracerProvider {
     }
   }
 
+  forceFlush(): Promise<void> {
+    const promises = this._registeredSpanProcessors.map(
+      (spanProcessor: SpanProcessor) => {
+        return spanProcessor.forceFlush();
+      }
+    );
+
+    return Promise.all(promises)
```

Comment:
According to spec I think it should be handled a bit different. Each `spanProcessor.forceFlush()` should be run in separate promise, then each promise should be resolved with either success or error. You should return the statuses for all of the span processors then.
Promise.all will be terminated on first error and you will not be able to check what happened with all processors, so I think it should wait till the end.

The function could look like this:

```
  forceFlush(): Promise<void> {
    /**
     * probably config needs to be extended with option for this, by default it
     * can be 30s (name exportTimeoutMillis)
     */
    const timeout = 30000; 
    const promises = this._registeredSpanProcessors.map(
      (spanProcessor: SpanProcessor) => {
        return new Promise((resolve) => {
          let resolved = false;
          const timer = setTimeout(()=> {
            resolve('timeout ....');
            resolved = true;
          }, timeout);
          spanProcessor.forceFlush().then(()=> {
            if (!resolved) {
              resolve(false);
            }
          }).catch(error=> {
            clearTimeout(timer);
            if (!resolved) {
              resolve(error);
            }
          })
        });
      }
    );
    return new Promise((resolve, reject) => {
      Promise.all(promises).then((results)=> {
        const errors = results.filter(result => result);
        if (errors.length > 0) {
          reject(errors);
        } else {
          resolve();
        }
      });
    });
  }
```

Reply:
Thank you! I have adapted the code to the above. Now trying to write some tests for it